src/gen_rules.ml

```diff
@@ -37,23 +46,30 @@ module Gen(P : Install_rules.Params) = struct
         { SC.Dir_with_dune. src_dir; ctx_dir; stanzas; scope; kind = dir_kind } =
     let for_stanza ~dir = function
       | Library lib ->
-        let cctx, merlin = Lib_rules.rules lib ~dir ~scope ~dir_contents ~dir_kind in
-        (Some merlin, Some (lib.buildable.loc, cctx))
+        let cctx, merlin =
+          Lib_rules.rules lib ~dir ~scope ~dir_contents ~dir_kind in
+        (Some merlin, Some (lib.buildable.loc, cctx), None)
```

Comment:
We are starting to return a lot of values here and it's becoming hard to remember what is what. What about introducing a record type?

Reply:
Sure, I introduced a record type to clean this up. I think we do need a record with a field for every stanza of the same type eventually. We keep doing this binning everywhere.